components/tidb_query/src/batch/runner.rs

```diff
@@ -440,16 +424,126 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
             }
 
             // Grow batch size
-            if batch_size < BATCH_MAX_SIZE {
-                batch_size *= BATCH_GROW_FACTOR;
-                if batch_size > BATCH_MAX_SIZE {
-                    batch_size = BATCH_MAX_SIZE
-                }
-            }
+            grow_batch_size(&mut batch_size);
```

Comment:
looks like you can grow batch size in `internal_handle_request` so that there is no need for a standalone function `grow_batch_size`.